fe/fe-core/src/main/java/org/apache/doris/job/base/AbstractJob.java

```diff
@@ -132,23 +153,16 @@ public void updateJobStatus(JobStatus newJobStatus) {
     protected abstract void checkJobParamsInternal();
 
     public static AbstractJob readFields(DataInput in) throws IOException {
-        // todo use RuntimeTypeAdapterFactory of Gson to do the serde
-        JobType jobType = JobType.valueOf(Text.readString(in));
-        switch (jobType) {
-            case INSERT:
-                return InsertJob.readFields(in);
-            case MTMV:
-                // return MTMVJob.readFields(in);
-                break;
-            default:
-                throw new IllegalArgumentException("unknown job type");
-        }
-        throw new IllegalArgumentException("unknown job type");
+        String jsonJob = Text.readString(in);
+        AbstractJob<?> job = GsonUtils.GSON.fromJson(jsonJob, AbstractJob.class);
+        job.setRunningTasks(new ArrayList<>());
```

Comment:
Why  need to set running tasks to empty?
The running tasks is not persisted.

Reply:
This is an initialization operation. When reading data from the metadata, this list should have been empty. However, due to potential reasons related to gson serialization, the result is null, thus displaying the initialization.